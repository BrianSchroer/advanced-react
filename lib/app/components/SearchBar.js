import React from 'react';
import PropTypes from 'prop-types';
import glamorous from 'glamorous';
import debounce from 'lodash.debounce';

const StyledInput = glamorous.input({
  marginBottom: '1em'
});

export default class SearchBar extends React.Component {

  static propTypes = {
    doSearch: PropTypes.func.isRequired
  }

  state = {
    searchTerm: ''
  }

  doSearch = debounce(() => { this.props.doSearch(this.state.searchTerm); }, 300);

  handleSearch = (event) => {
    this.setState({searchTerm: event.target.value}, this.doSearch);
  }

  render() {
    return (
      <StyledInput
        type="search"
        placeholder="Enter search term"
        value={this.state.searchTerm}
        onChange={this.handleSearch}
      />
    );
  }
}