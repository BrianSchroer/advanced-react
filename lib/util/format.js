export default class {
  static date(dateString) {
    return new Date(dateString).toDateString();
  }
}
