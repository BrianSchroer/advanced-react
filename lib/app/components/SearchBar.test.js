import React from 'react';
import { snapshotHelper } from '../../util/testHelpers';
import SearchBar from './SearchBar';

const testProps = {
  doSearch: () => {}
};

describe('SearchBar', () => {

  it('renders correctly', () => {
    snapshotHelper.assertMatch(<SearchBar {...testProps} />);
  });

});