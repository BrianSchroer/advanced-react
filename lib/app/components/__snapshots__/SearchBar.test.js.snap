// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<input
  className="css-28xugj"
  onChange={[Function]}
  placeholder="Enter search term"
  type="search"
  value=""
/>
`;
